wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -452,7 +452,7 @@ public FactoryParametersResolver getFactoryParametersResolver(Map<String, String
           return factoryParametersResolver;
         }
       }
-      return defaultFactoryResolver;
+      return defaultFactoryResolver.accept(parameters) ? defaultFactoryResolver : null;
```

Comment:
Can you adjust javadoc and explain that this method can return null and why?

Reply:
Yes, writing test and jdoc.